src/test/java/BaseTest.java

```diff
@@ -1,18 +1,161 @@
+import com.beust.jcommander.Parameter;
 import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
 import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.interactions.Actions;
 import org.openqa.selenium.safari.SafariDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
 import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.BeforeSuite;
-
-
+import org.testng.annotations.Parameters;
+import org.openqa.selenium.interactions.Actions;
+import java.time.Duration;
 public class BaseTest {
 
+    public static WebDriver driver = null;
+     WebDriverWait wait;
     @BeforeSuite
     static void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
+    public static Actions actions = null;
+
+    @BeforeMethod
+    @Parameters("BaseUrl")
+    public void setUpBrowser(String BaseUrl) {
+       // driver = new ChromeDriver();
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--disable-notifications","--remote-allow-origins=*", "--incognito","--start-maximized");
+        driver = new ChromeDriver(options);
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(5));
+        //driver.manage().window().maximize();
+        driver.get(BaseUrl);
+        wait = new WebDriverWait(driver,Duration.ofSeconds(5));
+         actions = new Actions(driver);
+    }
+
+   protected void openBrowser() {String url = "https://bbb.testpro.io/";
+       driver.get(url);
+   }
+
+    public void closeBrowser() {
```

Comment:
The @AfterMethod annotated method will be invoked after the execution of each test method. 

```
   @AfterMethod
    public void closeBrowser() {
        driver.quit();
    }
```